### PR TITLE
ENT-4423 Add Web RBAC documentations

### PR DESCRIPTION
**Pull request by aleksandrychev**
Merge together:
https://github.com/cfengine/mission-portal/pull/759
https://github.com/cfengine/nova/pull/1357

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@nickanderson @craigcomstock please review

File: enterprise-cfengine-guide/settings.markdown [hunk 1]
```diff
@@ -196,6 +196,33 @@ Mission portal can authenticate against an external directory.
 
 **See Also:** [LDAP authentication REST API][LDAP authentication API]
 
+
+## Role based access control ##
+
+![Role based access control](role_based_access_control_settings.png)
+
+Roles in Mission portal can be restricted to perform only configured actions. 
+And you can configure they on Role based access control settings page.
```
Comment (nickanderson) [inline enterprise-cfengine-guide/settings.markdown]: s/And y/Y/

Comment (nickanderson) [reply enterprise-cfengine-guide/settings.markdown]: Actually, second sentence: Configure role-based access controls from settings.

File: enterprise-cfengine-guide/settings.markdown [hunk 2]
```diff
@@ -196,6 +196,33 @@ Mission portal can authenticate against an external directory.
...
+
+**Special Notes:** 
+
+- Admin role has all permissions by default.
+
+- Cf_remoteagent role has all permissions related to API by default.
+
+- User with several roles will be permitted to perform actions from every role.
```
Comment (nickanderson) [inline enterprise-cfengine-guide/settings.markdown]: I would reword this slightly:

Permissions granted by roles are additive, users with multiple roles are permitted to perform actions granted by each role.

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+## Get all permissions list
+
+**URI:** https://hub.cfengine.com/api/rbac
+
+**Method:** GET
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
```
Comment (nickanderson) [inline reference/enterprise-api-ref/web-rbac.markdown]: I cry every time I see us do this in docs. I can't wait until we automate the docs for APIs.

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+  -X GET \
+  https://hub.cfengine.com/api/rbac
+'
+```
```
Comment (craigcomstock) [inline reference/enterprise-api-ref/web-rbac.markdown]: trailing single tick should be removed right? (')

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+**Method:** GET
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/rbac/user-permissions
+'
+```
```
Comment (craigcomstock) [inline reference/enterprise-api-ref/web-rbac.markdown]: trailing single tick (')

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    Role name
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/role/admin/permissions
+'
+```
```
Comment (craigcomstock) [inline reference/enterprise-api-ref/web-rbac.markdown]: trailing single tick (')

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+**Example request (curl):**
+ 
+```
+curl -k --user <username>:<password> \
+  -X POST \
+  https://hub.cfengine.com/api/role/role_name/permissions \
+  -H 'content-type: application/json' \
+  -d '["Inventory.post", "VariablesDictionary.get"]'
+'
+```
```
Comment (craigcomstock) [inline reference/enterprise-api-ref/web-rbac.markdown]: trailing single tick (')

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+**Example request (curl):**
+ 
+```
+curl -k --user <username>:<password> \
+  -X PUT \
+  https://hub.cfengine.com/api/role/role_name/permissions \
+  -H 'content-type: application/json' \
+  -d '["Inventory.post", "VariablesDictionary.get"]'
+'
+```
```
Comment (craigcomstock) [inline reference/enterprise-api-ref/web-rbac.markdown]: trailing single tick (')

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+**Example request (curl):**
+ 
+```
+curl -k --user <username>:<password> \
+  -X DELETE \
+  https://hub.cfengine.com/api/role/role_name/permissions \
+  -H 'content-type: application/json' \
+  -d '["Inventory.post", "VariablesDictionary.get"]'
+'
+```
```
Comment (craigcomstock) [inline reference/enterprise-api-ref/web-rbac.markdown]: trailing single tick (')

File: reference/enterprise-api-ref/web-rbac.markdown [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+
+**URI:** https://hub.cfengine.com/api/rbac
+
+**Method:** GET
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X GET \
```
[the remark this answers is not in the document]
Comment (aleksandrychev) [reply reference/enterprise-api-ref/web-rbac.markdown]: it's an example, I don't think that we need to specify these credentials. we use this structure in other API documentations.

File: reference/enterprise-api-ref/web-rbac.markdown (hunk 4 above)
Comment (aleksandrychev) [reply reference/enterprise-api-ref/web-rbac.markdown]: done. thanks

File: enterprise-cfengine-guide/settings.markdown (hunk 1 above)
Comment (aleksandrychev) [reply enterprise-cfengine-guide/settings.markdown]: done, thanks

File: enterprise-cfengine-guide/settings.markdown [hunk 11]
```diff
@@ -196,6 +196,33 @@ Mission portal can authenticate against an external directory.
...
 **See Also:** [LDAP authentication REST API][LDAP authentication API]
 
+
+## Role based access control ##
+
+![Role based access control](role_based_access_control_settings.png)
+
+Roles in Mission portal can be restricted to perform only configured actions. 
+And you can configure they on Role based access control settings page.
+
```
[the remark this answers is not in the document]
Comment (aleksandrychev) [reply enterprise-cfengine-guide/settings.markdown]: done

File: reference/enterprise-api-ref/web-rbac.markdown (hunk 6 above)
Comment (aleksandrychev) [reply reference/enterprise-api-ref/web-rbac.markdown]: fixed

**Comment by aleksandrychev** [comment]
@craigcomstock @nickanderson thank you for the review, I made new changes.

**Comment by craigcomstock** [comment]
@nickanderson are your comments resolved? For some reason github thinks they are not.

**Comment by nickanderson** [comment]
@craigcomstock yep, they are.

**Comment by craigcomstock** [comment]
@aleksandrychev looks like builds are failing for mission-portal and nova. When you can, give some thoughts on if we can merge these?

**Comment by aleksandrychev** [comment]
@craigcomstock  mission portal fails because nova PR isn't merged and there is needed SQL schema changes. Travis is green on Nova, Jenkins is red but redness not related to changes.
